repos/system_upgrade/el7toel8/actors/commonleappdracutmodules/files/dracut/85sys-upgrade-redhat/mount_usr.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/sh
+# -*- mode: shell-script; indent-tabs-mode: nil; sh-basic-offset: 4; -*-
+# ex: ts=8 sw=4 sts=4 et filetype=sh
+
+type info >/dev/null 2>&1 || . /lib/dracut-lib.sh
+
+filtersubvol() {
+    local _oldifs
+    _oldifs="$IFS"
+    IFS=","
+    set $*
+    IFS="$_oldifs"
+    while [ $# -gt 0 ]; do
+        case $1 in
+            subvol\=*) :;;
+            *) printf '%s' "${1}," ;;
+        esac
+        shift
+    done
+}
+
+mount_usr()
+{
+    local _dev _mp _fs _opts _rest _usr_found _ret _freq _passno
+    # check, if we have to mount the /usr filesystem
+    while read _dev _mp _fs _opts _freq _passno; do
+        [ "${_dev%%#*}" != "$_dev" ] && continue
+        if [ "$_mp" = "/usr" ]; then
+            case "$_dev" in
+                LABEL=*)
+                    _dev="$(echo $_dev | sed 's,/,\\x2f,g')"
+                    _dev="/dev/disk/by-label/${_dev#LABEL=}"
+                    ;;
+                UUID=*)
+                    _dev="${_dev#block:}"
+                    _dev="/dev/disk/by-uuid/${_dev#UUID=}"
+                    ;;
+            esac
+            if strstr "$_opts" "subvol=" && \
+                [ "${root#block:}" -ef $_dev ] && \
+                [ -n "$rflags" ]; then
+                # for btrfs subvolumes we have to mount /usr with the same rflags
+                rflags=$(filtersubvol "$rflags")
+                rflags=${rflags%%,}
+                _opts="${_opts:+${_opts},}${rflags}"
+            elif getargbool 0 ro; then
+                # if "ro" is specified, we want /usr to be mounted read-only
+                _opts="${_opts:+${_opts},}ro"
+            elif getargbool 0 rw; then
+                # if "rw" is specified, we want /usr to be mounted read-write
+                _opts="${_opts:+${_opts},}rw"
+            fi
+            echo "$_dev ${NEWROOT}${_mp} $_fs ${_opts} $_freq $_passno"
+            _usr_found="1"
+            break
+        fi
+    done < "$NEWROOT/etc/fstab" >> /etc/fstab
+
+    if [ "$_usr_found" != "" ]; then
+        info "Mounting /usr with -o $_opts"
+        mount "$NEWROOT/usr" 2>&1 | vinfo
+        mount -o remount,rw "$NEWROOT/usr"
+
+        if ! ismounted "$NEWROOT/usr"; then
+            warn "Mounting /usr to $NEWROOT/usr failed"
+            warn "*** Dropping you to a shell; the system will continue"
+            warn "*** when you leave the shell."
+            action_on_fail
+        fi
+    fi
+}
+
+if [ -f "$NEWROOT/etc/fstab" ]; then
```

Comment:
@vinzenz Are we sure that we will not hit again issue with `$NEWROOT`? Especially when the script is processed before `do-upgrade.sh`. Pls add the `$NEWROOT` fix to the top of the file as well:
```shell
export NEWROOT=${NEWROOT:-"/sysroot"}
```

Reply:
It looks good to me otherwise. Just add the line to ensure we will not reanimate buried rabbit :-)